tests/unit/spritelist/test_spritesequence.py

```diff
@@ -0,0 +1,26 @@
+import arcade
+
+class _CustomSpriteSolidColor(arcade.SpriteSolidColor):
+    pass
+
+def test_collective_draw(window: arcade.Window) -> None:
+    sprite_list1: arcade.SpriteList[arcade.Sprite] = arcade.SpriteList()
+    sprite_list1.append(arcade.SpriteSolidColor(16, 16, color=(255, 0, 0, 1)))
+
+    sprite_list2: arcade.SpriteList[_CustomSpriteSolidColor] = arcade.SpriteList()
+    sprite_list2.append(_CustomSpriteSolidColor(16, 16, color=(255, 0, 0, 1)))
+
+    # It really is a SpriteList with a good type; this would not typecheck otherwise
+    custom_sprite: _CustomSpriteSolidColor = sprite_list2[0] # assert_type
```

Comment:
Are you running this type check manually? My understanding is that we don't run type checks against the test folder at all. See:

https://github.com/pythonarcade/arcade/blob/94c42806cf63df636e2ae6eb2e8a0ceb25e31225/pyproject.toml#L140-L150

Reply:
I don't think I ran pyright, but I did run mypy manually.

Reply:
We auto-run it in our side for build CI. As an important note, mypy refuses to support for `NamedTuple` at all. Since pyglet uses `NamedTuple` in pyglet's `math.Vec*` and `Mat*` classes:
* they mark `pyglet.math` as an excluded module in their mypy config
* we use pyright